kafka-long-running-session-window-test/src/test/java/com/hazelcast/jet/tests/kafka/LongRunningKafkaSessionWindowTest.java

```diff
@@ -134,48 +125,33 @@ public void kafkaTest() throws InterruptedException {
         }
     }
 
-    private DAG testDAG() {
-        DAG dag = new DAG();
-        WatermarkGenerationParams<Trade> wmGenParams = wmGenParams(Trade::getTime, limitingLag(lagMs),
-                emitByMinStep(lagMs), 10_000);
-        Vertex readKafka = dag.newVertex("read-kafka", streamKafkaP(kafkaProps,
-                consumerRecord -> (Trade) consumerRecord.value(), wmGenParams, topic));
-
-        DistributedFunction<Trade, String> keyFn = Trade::getTicker;
-        DistributedToLongFunction<Trade> timestampFn = Trade::getTime;
-        Vertex aggregateSessionWindow = dag.newVertex("session-window",
-                aggregateToSessionWindowP(sessionTimeout, singletonList(timestampFn),
-                        singletonList(keyFn),
-                        counting(), WindowResult::new));
-
-        Vertex writeResultsToKafka = dag.newVertex("write-results-kafka",
-                writeKafkaP(kafkaPropertiesForResults(brokerUri, offsetReset), topic + "-result",
-                        (DistributedFunction<WindowResult<String, Long>, String>) WindowResult::getKey,
-                        (DistributedFunction<WindowResult<String, Long>, Long>) WindowResult::getStart));
-
-        dag
-                .edge(between(readKafka, aggregateSessionWindow)
-                        .partitioned(Trade::getTicker, HASH_CODE)
-                        .distributed()
-                )
-                .edge(from(aggregateSessionWindow).to(writeResultsToKafka));
-
-        return dag;
-    }
+    private Pipeline pipeline() {
+        Pipeline pipeline = Pipeline.create();
 
+        Properties propsForResult = kafkaPropertiesForResults(brokerUri, offsetReset);
 
-    private DAG verificationDAG() {
-        DAG dag = new DAG();
+        pipeline.drawFrom(KafkaSources.<String, Trade>kafka(kafkaProps, topic))
+                .map(Map.Entry::getValue)
+                .addTimestamps(Trade::getTime, lagMs)
+                .window(session(sessionTimeout))
+                .groupingKey(Trade::getTicker)
+                .aggregate(counting())
+                .drainTo(KafkaSinks.kafka(propsForResult, topic + "-results"));
+
+        return pipeline;
+    }
 
-        Vertex readVerificationRecords = dag.newVertex("read-verification-kafka",
-                streamKafkaP(kafkaPropertiesForResults(brokerUri, offsetReset), noWatermarks(), topic + "-result"));
+    private Pipeline verificationPipeline() {
+        Pipeline pipeline = Pipeline.create();
 
-        long countCopy = countPerTicker;
-        Vertex verifyRecords = dag.newVertex("verification", () -> new VerificationSink(countCopy));
+        Properties properties = kafkaPropertiesForResults(brokerUri, offsetReset);
+        final int countCopy = countPerTicker;
+        pipeline.drawFrom(KafkaSources.kafka(properties, topic + "-results"))
+                .drainTo(Sinks.fromProcessor("verification",
```

Comment:
I think you can use the custom sink builder here too

Reply:
👍